### PR TITLE
Run benchmarks using example sidekiq application

### PR DESCRIPTION
**Pull request by pawelchcki**
I've cleaned up and made CI jobs out of sample Sidekiq application I was using for benchmarking memory use.

In current configuration it runs two tasks on the CI:
1) runs the benchmark with all instrumentations enabled
2) runs the benchmark with all instrumentations disabled

Next steps could be to optimize the presentation of both results, i.e. graph both datapoints on one graph and add it as an artifact to the build so We can visually see the difference between two runs.

Additionally we could calculate median and 95th percentile memory consumption and use that create a impact metric that we then could use to compare and reference in future performance improvements.

One additional metric we got from these benchmarks its the runtime. It seems it takes about 20% less time to complete the benchmark when running with instrumentations disabled.

### CODE REVIEW

**Comment by delner** [comment]
I really like this concept! Let me take a deeper dive into it and get back to you on this.

File: benchmarks/sidekiq_test.rb [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+end
+
+class Sample < ActiveRecord::Base; end
+
+require 'sidekiq/launcher'
+require 'sidekiq/cli'
+require 'concurrent/atomic/atomic_fixnum'
+
+Sidekiq.configure_server do |config|
+  redis_conn = proc { Redis.new(host: 'localhost', port: 6379) }
```
Comment (delner) [inline benchmarks/sidekiq_test.rb]: Does this need to be configurable? Redis might not be running on the localhost, particularly in a Docker environment.

File: benchmarks/sidekiq_test.rb [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+    Sample.last(100).to_a
+  end
+end
+
+if Datadog.respond_to?(:configure)
+  Datadog.configure do |d|
+    d.use :rails,
+          enabled: true,
+          auto_instrument_redis: true,
```
Comment (delner) [inline benchmarks/sidekiq_test.rb]: This is old syntax: does not work. Should be updated to new configuration.

Comment (pawelchcki) [reply benchmarks/sidekiq_test.rb]: in new configuration this is Noop - as its enabled by default. I've remove these lines not to confuse anyone. It was hover useful when benchmarking older versions

**Comment by pawelchcki** [comment]
I've addressed your comments @delner. Should be ready for next round/merging :)

**Pull request merged**